lib/OpenQA/Schema/Result/ScheduledProducts.pm

```diff
@@ -528,44 +528,28 @@ sub _generate_jobs {
            # note: That order also defines the precedence from lowest to highest. The only exception is the WORKER_CLASS
            #       variable where all occurrences are merged.
             my %settings;
-            my @worker_classes;
-            for my $entity ($product, $job_template->machine, $job_template->test_suite, $job_template) {
-                my %settings_of_entity = map { $_->key => $_->value } $entity->settings;
-                if (my $worker_class = delete $settings_of_entity{WORKER_CLASS}) {
-                    push(@worker_classes, $worker_class);
-                }
-                @settings{keys %settings_of_entity} = values %settings_of_entity;
-            }
+            my %params = (
+                settings     => \%settings,
+                input_args   => $args,
+                product      => $product,
+                machine      => $job_template->machine,
+                test_suite   => $job_template->test_suite,
+                job_template => $job_template,
+            );
+            my $error = OpenQA::JobSettings::generate_settings(\%params);
+            $error_message .= $error if defined $error;
 
             # add properties from dedicated database columns to settings
             $settings{TEST}            = $job_template->name || $job_template->test_suite->name;
-            $settings{MACHINE}         = $job_template->machine->name;
-            $settings{BACKEND}         = $job_template->machine->backend;
             $settings{TEST_SUITE_NAME} = $job_template->test_suite->name;
             $settings{JOB_DESCRIPTION} = $job_template->description if length $job_template->description;
 
-            # merge worker classes
-            $settings{WORKER_CLASS} = @worker_classes ? join(',', sort(@worker_classes)) : "qemu_$args->{ARCH}";
-
-            # add upper-case versions of keys
-            for my $key (keys %$args) {
-                next if $key eq 'TEST' || $key eq 'MACHINE';
-                $settings{uc $key} = $args->{$key};
-            }
-
             # make sure that the DISTRI is lowercase
             $settings{DISTRI} = lc($settings{DISTRI});
```

Comment:
FYI, this PR caused regression in many tests that use `DISTRI` in asset names or job dependency settings. The above line must be executed before `generate_settings()`.

Reply:
Example: https://openqa.suse.de/tests/4186731#settings